deploy/mlflow-triton-plugin/mlflow_triton/config.py

```diff
@@ -24,17 +24,114 @@
+        # Must handle paths with s3 prefix
+        start = s3_path.find("s3://")
+        path = ""
+        if start != -1:
+            path = s3_path[start + len("s3://"):]
+            clean_path = ("s3://")
+        else:
+            path = s3_path
+            clean_path = ""
+
+        # Must handle paths with https:// or http:// prefix
+        https_start = path.find("https://")
+        if https_start != -1:
+            path = path[https_start + len("https://"):]
+            clean_path += "https://"
+        else:
+            http_start = path.find("http://")
+            if http_start != -1:
+                path = path[http_start + len("http://"):]
+                clean_path += "http://"
+
+        # Remove trailing slashes
+        rtrim_length = len(path.rstrip('/'))
+        if rtrim_length == 0:
+            raise MlflowException("Invalid bucket name: '" + path + "'")
+
+        # Remove leading slashes
+        ltrim_length = len(path) - len(path.lstrip('/'))
+        if ltrim_length == len(path):
+            raise MlflowException("Invalid bucket name: '" + path + "'")
+
+        # Remove extra internal slashes
+        true_path = path[ltrim_length:rtrim_length + 1]
+        slash_locations = []
+        previous_slash = False
+        for i in range(len(true_path)):
+            if true_path[i] == '/':
+                if not previous_slash:
+                    clean_path += true_path[i]
+                previous_slash = True
+            else:
+                clean_path += true_path[i]
+                previous_slash = False
+
+        return clean_path
+
     def __init__(self):
         super().__init__()
+        self.s3_regex = re.compile(
+            's3://(http://|https://|)([0-9a-zA-Z\\-.]+):([0-9]+)/'
+            '([0-9a-z.\\-]+)(((/[0-9a-zA-Z.\\-_]+)*)?)')
         self['triton_url'] = os.environ.get('TRITON_URL')
         self['triton_model_repo'] = os.environ.get('TRITON_MODEL_REPO')
 
         if self['triton_model_repo'].startswith('s3://'):
+            uri = self.parse_path(self['triton_model_repo'])
+            if uri.protocol == "https://":
+                scheme = "https://"
+            else:
+                scheme = "http://"
+            endpoint_url = None
+            if uri.host_name != "" and uri.host_port != "":
+                endpoint_url = '{}{}:{}'.format(
+                    scheme, uri.host_name, uri.host_port)
             import boto3
-            self['s3'] = boto3.client('s3')  # boto3 handles AWS credentials
-            self['s3_bucket'] = self['triton_model_repo'].replace('s3://', '').replace('/', '')
+            # boto3 handles AWS credentials
+            self['s3'] = boto3.client(
+                's3', endpoint_url=endpoint_url)
+            self['s3_bucket'] = uri.bucket
+            self['s3_prefix'] = uri.prefix
+            self['triton_model_repo'] = 's3://{}'.format(
+                '/'.join(filter(None, [uri.bucket, uri.prefix])))
```

Comment:
What is the point of the `filter()` here?

Reply:
`os.path.join` works just fine. Please see the latest commit, thanks.